pkg/cli/mirror/list/updates.go

```diff
@@ -25,29 +43,199 @@ func NewUpdatesCommand(f kcmdutil.Factory, ro *cli.RootOptions) *cobra.Command {
 			oc-mirror list updates --config mirror-config.yaml
 		`),
 		Run: func(cmd *cobra.Command, args []string) {
-			kcmdutil.CheckErr(o.Complete(cmd, f, args))
 			kcmdutil.CheckErr(o.Validate())
-			kcmdutil.CheckErr(o.Run())
+			kcmdutil.CheckErr(o.Run(cmd.Context()))
 		},
 	}
 
 	o.BindFlags(cmd.PersistentFlags())
 
 	fs := cmd.Flags()
 	fs.StringVarP(&o.ConfigPath, "config", "c", o.ConfigPath, "Path to imageset configuration file")
-
 	return cmd
 }
 
-func (o *UpdatesOptions) Complete(cmd *cobra.Command, f kcmdutil.Factory, args []string) error {
+func (o *UpdatesOptions) Validate() error {
+	if len(o.ConfigPath) == 0 {
+		return fmt.Errorf("must specify config using --config")
+	}
 	return nil
 }
 
-func (o *UpdatesOptions) Validate() error {
+func (o *UpdatesOptions) Run(ctx context.Context) error {
+	cfg, err := config.LoadConfig(o.ConfigPath)
+	if err != nil {
+		return err
+	}
+
+	path := filepath.Join(o.Dir, config.SourceDir)
+	backend, err := storage.ByConfig(ctx, path, cfg.StorageConfig)
+	if err != nil {
+		return fmt.Errorf("error opening backend: %v", err)
+	}
+
+	var meta v1alpha1.Metadata
+	switch err := backend.ReadMetadata(ctx, &meta, config.MetadataBasePath); {
+	case err != nil && !errors.Is(err, storage.ErrMetadataNotExist):
+		return err
+	case err != nil && errors.Is(err, storage.ErrMetadataNotExist):
+		return fmt.Errorf("no metadata detected")
+	case err == nil && len(meta.PastMirrors) != 0:
+		if len(cfg.Mirror.OCP.Channels) != 0 {
+			if err := o.releaseUpdates(ctx, cfg, meta); err != nil {
+				return err
+			}
+		}
+		if len(cfg.Mirror.Operators) != 0 {
+			if err := o.operatorUpdates(ctx, cfg, meta); err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+func (o UpdatesOptions) releaseUpdates(ctx context.Context, cfg v1alpha1.ImageSetConfiguration, meta v1alpha1.Metadata) error {
+	uuid := uuid.New()
+	// QUESTION(jpower): Handle multiple arch?
+	arch := "amd64"
+	logrus.Info("Getting release update information")
+	for _, ch := range cfg.Mirror.OCP.Channels {
+		url := cincinnati.UpdateUrl
+		if ch.Name == "okd" {
+			url = cincinnati.OkdUpdateURL
+		}
+
+		client, upstream, err := cincinnati.NewClient(url, uuid)
+		if err != nil {
+			return err
+		}
+
+		// Find the last release downloads if no downloads
+		// have been made in the target channel list
+		// all versions
+		var vers []semver.Version
+		ver, err := cincinnati.FindLastRelease(meta, ch.Name, url, uuid)
+		switch {
+		case err != nil && !errors.As(err, &cincinnati.ErrNoPreviousRelease):
```

Comment:
Use `As` when the error is typed, `Is` otherwise
```suggestion
		case err != nil && !errors.Is(err, cincinnati.ErrNoPreviousRelease):
```

Reply:
Thanks! This has been fixed on the branch so I am going to resolve this thread.